### PR TITLE
Mask conversion

### LINKED ISSUE #41: Use `skimage.measure.find_contours` for mask to contour conversion
https://scikit-image.org/docs/stable/api/skimage.measure.html#find-contours

Need to handle the case where open contours are created. Especially at the corner of the image.

### PR DESCRIPTION
**Pull request by SimonBiggs**
Creates functions to convert between contours defined by coordinates and masks. Of note, the masks used by `radiotherapyai` are not binary masks, instead, edge pixels are scaled proportionally depending on how much of a given pixel is within the contour. 

Fixes #41

This PR does not address the case when contours are nested as in https://dicom.nema.org/medical/Dicom/2022b/output/chtml/part03/sect_C.8.8.6.3.html. A follow up PR will pull in the keyhole technique.

### CODE REVIEW

File: radiotherapyai/metrics/dice.py
Comment (sjswerdloff) [inline radiotherapyai/metrics/dice.py:40]: it wouldn't hurt to identify the expected limits [0,1] and a ten words or left description of what it means (overlap).
Links are good for someone who would like more detail.
But most people are already hip deep in code, and if they are, wandering off for a deep dive to figure out "what is this function" is kind of a task...

**Review by sjswerdloff — APPROVED** [review]

**Comment by sjswerdloff** [comment]
> I'd be interested in some ideas about how to make the tests a bit more obvious to read for geometric cases like this.
> 
> Alternatively I could make some tests based on PNG files... That way the results can also be visually inspected... Hmm.
> 
PNGs just showing what each test is would be fine, even if you don't use them as the basis.
Or for small grids, putting numbers in the grid to show what's going on.
> Also, happy to take feedback on the review process. Feel free (as always) to let me know if the PR is too big etc.
This PR is a bit too big.  But if there was a visual representation of the key algorithms and tests it would have reduce the cognitive load to "not quite too big"

**Comment by sjswerdloff** [comment]
whoops, didn't mean to close

File: radiotherapyai/metrics/dice.py
Comment (SimonBiggs) [reply radiotherapyai/metrics/dice.py:40]: Yup, good call. Will fix.

**Comment by SimonBiggs** [comment]
> PNGs just showing what each test is would be fine, even if you don't use them as the basis.

I like it. Will do. I'll fix before merge.

File: radiotherapyai/metrics/dice.py
Comment (SimonBiggs) [reply radiotherapyai/metrics/dice.py:40]: > I'll fix before merge.

Actually, would you prefer I merge this and then make those changes in a new PR so that the new changes are easier to review?

Or is it much of a muchness, and best to have it in this PR and then re-request your approval?

Comment (sjswerdloff) [reply radiotherapyai/metrics/dice.py:40]: New pr please

**Pull request merged**
